components/google-cloud/google_cloud_pipeline_components/experimental/custom_job/custom_job_sample.ipynb

```diff
@@ -0,0 +1,603 @@
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Create a Cloud Storage bucket as necessary\n",
+    "\n",
+    "You need a Cloud Storage bucket for this example.  If you don't have one that you want to use, you can make one now.\n",
+    "\n",
+    "\n",
+    "Set the name of your Cloud Storage bucket below. It must be unique across all\n",
+    "Cloud Storage buckets.\n",
+    "\n",
+    "You can also change the `REGION` variable, which is used for operations\n",
+    "throughout the rest of this notebook. Make sure to [choose a region where Vertex AI services are\n",
+    "available](https://cloud.google.com/vertex-ai/docs/general/locations#available_regions). You may\n",
+    "not use a Multi-Regional Storage bucket for training with Vertex Pipelines."
+   ],
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "BUCKET_NAME = \"gs://[your-bucket-name]\"  # @param {type:\"string\"}\n",
+    "REGION = \"us-central1\"  # @param {type:\"string\"}"
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "if BUCKET_NAME == \"\" or BUCKET_NAME is None or BUCKET_NAME == \"gs://[your-bucket-name]\":\n",
+    "    BUCKET_NAME = \"gs://\" + PROJECT_ID + \"aip-\" + TIMESTAMP"
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "**Only if your bucket doesn't already exist**: Run the following cell to create your Cloud Storage bucket."
+   ],
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "! gsutil ls -al $BUCKET_NAME"
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Finally, validate access to your Cloud Storage bucket by examining its contents:"
+   ],
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "! gsutil ls -al $BUCKET_NAME"
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Import libraries and define constants"
+   ],
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "import google_cloud_pipeline_components as gcpc\n",
+    "from kfp.v2 import components, compiler, dsl"
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "PATH=%env PATH\n",
+    "%env PATH={PATH}:/home/jupyter/.local/bin\n",
+    "\n",
+    "USER = \"your-user-name\"  # <---CHANGE THIS\n",
+    "PIPELINE_ROOT = \"{}/pipeline_root/{}\".format(BUCKET_NAME, USER)\n",
+    "\n",
+    "PIPELINE_ROOT"
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## Create a component, convert it to a custom job to use in a pipeline.\n",
```

Comment:
a great portion of this notebook is common setup, and useful for all FPC. I think we should consider have a common setup notebook at some root directory, and point to that notebook from here, for 2 purpose
1. make this sample very clear and concise on how to use this specific component
2. make other authors write FPC samples easier.

Reply:
completely agree with that, this was based on DevRel prior work. I will simplify this notebook to the core concepts and create a tracker for a general setup  instructions notebook.